### PR TITLE
[Tensor] ShortTensor class with unsigned 16-bit integer 

### PR DESCRIPTION
**Pull request by djeong20**
In this PR, a new type of tensor, the ShortTensor class, is designed explicitly for handling unsigned 16-bit integer data types.
This new tensor class aims to provide users with more options when working with various data types.
Note that the ShortTensor class does not support mathematical operations like multiplication or addition.

**Self-evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test:   [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Donghyeon Jeong <dhyeon.jeong@samsung.com>

### CODE REVIEW

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Review by skykongkong8 — APPROVED** [review]
LGTM!

File: nntrainer/tensor/short_tensor.h [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+class ShortTensor : public TensorBase {
+public:
+  /**
+   * @brief     Basic Constructor of Tensor
+   */
+  ShortTensor(std::string name_ = "", Tformat fm = Tformat::NCHW);
+
+  /**
+   * @brief Construct a new ShortTensor object
+   *
```
Comment (SeoHyungjun) [inline nntrainer/tensor/short_tensor.h:32]: Is a blank line required?

File: nntrainer/tensor/short_tensor.cpp [hunk 2]
```diff
@@ -0,0 +1,365 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * @file	short_tensor.cpp
+ * @date	02 April 2024
```
Comment (SeoHyungjun) [inline nntrainer/tensor/short_tensor.cpp:4]: Isn't it July or August?

**Review by SeoHyungjun — APPROVED** [review]
LGTM

File: nntrainer/tensor/short_tensor.cpp [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+}
+
+uint16_t &ShortTensor::getValue(unsigned int b, unsigned int c, unsigned int h,
+                                unsigned int w) {
+  return getValue(getIndex(b, c, h, w));
+}
+
+void ShortTensor::setValue(float value) {
+  uint16_t *data = (uint16_t *)getData();
+  std::fill(data, data + size(), value);
```
Comment (EunjuYang) [inline nntrainer/tensor/short_tensor.cpp:180]: Is it ok to use `float` data type without casting?

**Review by EunjuYang — APPROVED** [review]
Overall LGTM !

File: nntrainer/tensor/short_tensor.cpp (hunk 3 above)
Comment (skykongkong8) [reply nntrainer/tensor/short_tensor.cpp:180]: Nice catch!
I also think it needs fixing

**Pull request merged**